extensions/sql-database-projects/src/tools/buildHelper.ts

```diff
@@ -67,66 +69,114 @@ export class BuildHelper {
 		const fullSdkName = `${sdkName}.${sdkVersion}`;
 
 		// check if this if the nuget needs to be downloaded
-		const nugetPath = path.join(this.extensionBuildDir, `${fullSdkName}.nupkg`);
+		const buildSdkNugetPath = path.join(this.extensionBuildDir, `${fullSdkName}.nupkg`);
 
-		if (await utils.exists(nugetPath)) {
+		let missingDacFxFiles = false;
+		let missingScriptDom = false;
+
+		if (await utils.exists(buildSdkNugetPath)) {
```

Comment:
No need for this because the first iteration of the contained loop would trigger, right?

Reply:
this is so that the dlls are only checked if the correct version of the nuget is in the build folder, otherwise the new nuget might not get downloaded if an older version of the build dlls were previously downloaded. 

This doesn't currently handle when switching back to an older version of the DacFx nuget that was previously downloaded, since that wouldn't trigger a new download and replacing the dlls, so I'll add a comment to do that.